trio/_core/tests/tutil.py

```diff
@@ -80,6 +81,40 @@ def ignore_coroutine_never_awaited_warnings():
             gc_collect_harder()
 
 
+def _noop(*args, **kwargs):
+    pass
+
+
+if sys.version_info >= (3, 8):
+
+    @contextmanager
+    def restore_unraisablehook():
+        sys.unraisablehook, prev = sys.__unraisablehook__, sys.unraisablehook
+        try:
+            yield
+        finally:
+            sys.unraisablehook = prev
+
+    @contextmanager
+    def disable_threading_excepthook():
+        threading.excepthook, prev = _noop, threading.excepthook
+        try:
+            yield
+        finally:
+            threading.excephtook = prev
+
+
+else:
+
+    @contextmanager
+    def restore_unraisablehook():
```

Comment:
I could do this if it were prefered:

```python
def _noop(*args, **kwargs):
    pass

@contextmanager
def nullcontext():
    yield

assert nullcontext()  # force coverage

if sys.version_info >= (3, 8):

    @contextmanager
    def restore_unraisablehook():
        sys.unraisablehook, prev = sys.__unraisablehook__, sys.unraisablehook
        try:
            yield
        finally:
            sys.unraisablehook = prev

    @contextmanager
    def disable_threading_excepthook():
        threading.excepthook, prev = _noop, threading.excepthook
        try:
            yield
        finally:
            threading.excephtook = prev


else:
    restore_unraisablehook = disable_threading_excepthook = nullcontext
```

Reply:
That does seem maybe generally better, though I don't super like the `assert nullcontext()`.  I wouldn't hold up here for Trio to change to my quickly-thought-up `coverage unneeded`/`coverage wanted` strategy so maybe keep moving with `no cover` on the `nullcontext()` or a 'real' test that uses `nullcontext()` with a `with`.


```python
def test_nullcontext():
    with nullcontext() as null:
        assert null is None
```

Or somesuch...  easy to add right adjacent to the definition, not complicated, not too noisy I think, and it actually checks 'interesting' properties of the code (as opposed to the truthiness of the context manager as the `assert nullcontext()` did).

Reply:
@altendky I'm happy to go with any of these options

Reply:
Your proposal here plus the actual test seems good?